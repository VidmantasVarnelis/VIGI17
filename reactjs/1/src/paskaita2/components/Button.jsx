import React from 'react';

// Reacto komponentai yra kuriami PaascalCase
const Button = (props) => {
    return (
        <button
            onClick={props.onButtonClick}
            className={props.className ? props.className : 'primary_button'}
        >
            {props.text ? props.text : 'Click me!'}
        </button>
    );
};

export default Button;
