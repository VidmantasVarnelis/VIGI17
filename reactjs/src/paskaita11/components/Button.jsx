import React from 'react';
import styles from './button.module.scss';

const Button = ({ title, size, style, onClick, variant = 'primary' }) => {
  const buttonSizeClass = size === 'sm' ? styles['size-sm'] : styles['size-lg'];
  const variantBackgroundStyles =
    variant === 'primary'
      ? styles['primary-btn']
      : variant === 'danger' && styles['danger-btn'];

  return (
    <button
      className={`${variantBackgroundStyles} ${buttonSizeClass}`}
      style={{ ...style }}
      onClick={onClick}
    >
      {title}
    </button>
  );
};

export default Button;
