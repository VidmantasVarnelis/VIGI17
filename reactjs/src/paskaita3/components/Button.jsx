import React from 'react';

class Button extends React.Component {
    render() {
        return (
            <button>
                {this.props.title ? this.props.title : 'Default text!'}
            </button>
        );
    }
}
export default Button;
// Funkcinis atrodytu taip:

// const Button =(props) => {
//     return <button>{props.title}</button>
// }
