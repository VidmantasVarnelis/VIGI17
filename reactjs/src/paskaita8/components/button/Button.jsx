import React from 'react';
import styled from 'styled-components';

const StyledButton = styled.button`
  padding: 0.5rem 1rem;
  background-color: #4b4b4b;
  border: none;
  border-radius: 0.25rem;
  color: white;
  font-size: 1rem;
`;

const Button = ({ title }) => {
  return <StyledButton>{title}</StyledButton>;
};

export default Button;
