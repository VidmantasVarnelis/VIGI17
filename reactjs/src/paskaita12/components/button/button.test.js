import { render, screen } from '@testing-library/react';
import Button from './Button';

describe('Button tests', () => {
  //   it('Button renders corretly', () => {
  //     render(<Button title='Increment' />);
  //     expect(screen.getByRole('button')).toBeInTheDocument();
  //     expect(screen.getByRole('button').textContent).toBe('Increment');
  //   });
  it('Button renders corretly', () => {
    const { container } = render(<Button title='CodeAcademy' />);
    expect(container).toMatchSnapshot();
  });
});
