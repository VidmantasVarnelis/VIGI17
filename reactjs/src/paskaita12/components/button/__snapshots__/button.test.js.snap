// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests Button renders corretly 1`] = `
<div>
  <button>
    CodeAcademy
  </button>
</div>
`;
